pkg/cache/components/component_suite_test.go

```diff
@@ -0,0 +1,85 @@
+package components
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	cfg       *rest.Config
+	k8sClient kubernetes.Interface
+	testEnv   *envtest.Environment
+)
+
+func TestComponents(t *testing.T) {
+	RegisterFailHandler(Fail)
+	SetDefaultEventuallyTimeout(1 * time.Minute)
+	SetDefaultEventuallyPollingInterval(1 * time.Second)
+	RunSpecs(t, "Components Suite")
+}
+
+var _ = BeforeSuite(func() {
+	testEnv = &envtest.Environment{}
+
+	var err error
+
+	// cfg is defined in this file globally.
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	k8sClient, err = kubernetes.NewForConfig(cfg)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	// initialize a set of namespaces and pods
```

Comment:
I would actually summarize the code.
```suggestion
	// initialize 2 namespaces with 5 pods
```

I know that risks getting out of date, but I can actually SKIM the code after reading that comment.

Reply:
Makes sense. Added a comment in https://github.com/everettraven/telescopia/pull/14/commits/da1f64c076a3c3f721be56e3ac779ca229808298 that should resolve this